apps/api/src/conversations/conversations.gateway.ts

```diff
@@ -0,0 +1,292 @@
+import {
+  ClassSerializerInterceptor,
+  UseFilters,
+  UseInterceptors,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ConnectedSocket,
+  MessageBody,
+  OnGatewayConnection,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { AuthService } from '../auth/auth.service';
+import { ConversationsService } from './conversations.service';
+import { sendMessageDto } from './dtos/sendMessage.dto';
+import { isUUIDDto } from './dtos/IsUUID.dto';
+import { HttpExceptionTransformationFilter } from '../common/filters/HttpExceptionFilter.filter';
+import { User } from '../users/user.entity';
+import { createConversationDto } from './dtos/createConversation.dto';
+import { updateRoleDto } from './dtos/updateRole.dto';
+import { Conversation } from './entities/conversation.entity';
+import { muteUserDto } from './dtos/muteUser.dto';
+import { isDateDto } from './dtos/isDate.dto';
+import { conversationRestrictionEnum } from './conversationRestriction.enum';
+import { ConversationsDetails, unreadMessagesResponse } from 'types';
+import { Message } from './entities/message.entity';
+import { ConversationRole } from './entities/conversationRole.entity';
+import { UsersService } from '../users/users.service';
+import { instanceToPlain } from 'class-transformer';
+
+@UseFilters(HttpExceptionTransformationFilter)
+@UsePipes(new ValidationPipe())
+@UseInterceptors(ClassSerializerInterceptor)
+@WebSocketGateway({
+  namespace: 'conversations',
+})
+export class ConversationsGateway implements OnGatewayConnection {
+  @WebSocketServer() server!: Server;
+
+  constructor(
+    private readonly conversationsService: ConversationsService,
+    private readonly authService: AuthService,
+    private readonly usersService: UsersService,
+  ) {}
+
+  async handleConnection(client: Socket): Promise<string | void> {
+    console.log('New client trying to connect');
+    if (!client.handshake.headers?.authorization) {
+      client.disconnect();
+      return;
+    }
+    const currentUser = this.authService.verifyUserFromToken(
+      client.handshake.headers.authorization,
+    );
+    if (!currentUser) {
+      client.disconnect();
+      return;
+    }
+    client.data = { id: currentUser.sub, name: currentUser.name };
+    client.join(`user_${currentUser.sub}`);
+    const conversations = await this.conversationsService.getConversationsIds(
+      currentUser.sub,
+    );
+    conversations.forEach((el) => client.join(`conversation_${el}`));
+    return 'Connection established';
+  }
+
+  // handleDisconnect(client : any) {
+  // this.clients.get(client.data.id)?.delete(client.id)
+  // if (this.clients.get(client.data.id)?.size === 0) this.clients.delete(client.data.id)
+  // console.error("Removed socket", this.clients);
+  // }
```

Comment:
Should this be kept as a comment?

Reply:
Yes for now, we might have to do some cleanup on disconnection, I will know for sure when I implement it in the front